### ISSUE TITLE
Dotnet demo crahses with TrapException

### ISSUE THREAD

**Issue by pawel-kedzior-sw** [bytecodealliance/wasmtime-demos#14]
I am getting TrapException when trying to run dotnet demo.

```
Unhandled exception. Wasmtime.TrapException: trap during instantiation
   at Wasmtime.Instance..ctor(Module module, IHost host)
   at Wasmtime.Module.Instantiate(IHost host)
   at WasmtimeDemo.Program.Main() in C:\Users\Administrator\dev\wasmtime-demos\dotnet\Program.cs:line 20
```

I am following the instructions from:
https://github.com/bytecodealliance/wasmtime-demos/tree/master/dotnet

I have:
- wasmtime-demos 171b4cbaded
- Wasmtime.Dotnet.dll  0.8.0-preview2
- dotnet 3.1.100

It reproduced on two machines for me.

Any ideas?

**Comment by peterhuene** [comment]
Hi @zuraff.

There's some more context to the problem here: https://github.com/bytecodealliance/wasmtime/issues/753

Are you also reproducing this on Windows?

It appears that something has changed with the toolset such that the generated WebAssembly module can't be loaded by Wasmtime (a problem with living on the edge of new technology, sadly).

I haven't yet determined why this seems to only happen on Windows.

The .wasm file I attached in my [comment](https://github.com/bytecodealliance/wasmtime/issues/753#issuecomment-570363275) was built from macOS with the same toolset versions but doesn't appear to have the same problem.  Give that file a try to complete the demo walkthrough and let me know if it works.

I'll continue the investigation via the Wasmtime issue above, so I'm going to close this issue for now.

Thanks for reporting this to us!

**Comment by pawel-kedzior-sw** [comment]
Yes, Windows, the problem reproduces on these two for me:
* Windows 10 Enterprise
* Windows Server 2019 Standard

**Comment by pawel-kedzior-sw** [comment]
Fixed in 477a442c65
Many thanks.
